cunumeric/array.py

```diff
@@ -1959,9 +1959,12 @@ def compress(self, condition, axis=None, out=None):
                 category=RuntimeWarning,
             )
             condition = condition.astype(bool)
+
         if axis is None:
             axis = 0
             a = self.ravel()
+        else:
+            axis = normalize_axis_index(axis, self.ndim)
```

Comment:
Added because Numpy seems to accept negative axes:
```
In [6]: a = np.array([[1, 2], [3, 4], [5, 6]])

In [7]: np.compress([False, True], a, axis=-1)
Out[7]: 
array([[2],
       [4],
       [6]])
```